DESIGN.md

```diff
@@ -0,0 +1,27 @@
+## Overview
+
+### Target Audience
+
+- Developer A: high-level framework writer
+  - wants to provide barebones initial configuration
+  - wants to consume a simple event stream
+  - wants to inject events into a managed event loop (easy state machine semantics)
+
+```go
+c := client.New(connectionOptions..) // package scheduler/client
+eventCh, errCh := c.Register(registrationOptions..)
```

Comment:
Some more pointers in case we insist on using an event channel:

- https://github.com/mesosphere/mesos-dns/blob/master/detect/masters.go#L36
- https://www.quora.com/Are-all-Golang-libraries-non-blocking-and-async